server/src/main/java/org/elasticsearch/rest/BaseRestHandler.java

```diff
@@ -76,6 +79,16 @@ public final long getUsageCount() {
 
     @Override
     public final void handleRequest(RestRequest request, RestChannel channel, NodeClient client) throws Exception {
+        if (allowSystemIndexAccessByDefault() == false) {
+            final String allow_system_index_access = request.param("allow_system_index_access");
+            final ThreadContext threadContext = client.threadPool().getThreadContext();
+            if (threadContext.getHeader(SYSTEM_INDEX_ACCESS_CONTROL_KEY) == null
+                && (allow_system_index_access == null
+                || allow_system_index_access.isEmpty()
+                || Boolean.parseBoolean(allow_system_index_access) == false)) {
```

Comment:
Please use `Booleans.parseBoolean` instead, see #59190. This will also remove the need for the null/empty checks.